src/runtime/HalidePyTorchHelpers.h

```diff
@@ -0,0 +1,91 @@
+#ifndef HL_PYTORCH_WRAPPER_H
+#define HL_PYTORCH_WRAPPER_H
+
+/** \file
+ * Set of utility functions to wrap PyTorch tensors into Halide buffers,
+ * making sure the data in on the correct device (CPU/GPU).
+ */
+
+#include <exception>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+
+#include "torch/extension.h"
+
+#include "HalideBuffer.h"
+
+#ifdef HL_PT_CUDA
+#include "HalideRuntimeCuda.h"
+#include "cuda.h"
+#include "cuda_runtime.h"
+#endif
+
+#define WEAK __attribute__((weak))
```

Comment:
(1) It might be wise to choose a more unique macro name than WEAK here. HLPT_WEAK?
(2) I presume you realize that this will never build on Windows. (Does PyTorch? I have no idea)

Reply:
done

Reply:
Good point, I removed the WEAK macro